src/api/assets/assetsApi.ts

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { TokenList } from '@alephium/token-list'
+import { hexToString } from '@alephium/web3'
+import { createQueryKeyStore } from '@lukemorales/query-key-factory'
+
+import {
+  AssetBase,
+  NFTFile,
+  UnverifiedFungibleTokenMetadata,
+  UnverifiedNFTMetadata,
+  VerifiedFungibleTokenMetadata
+} from '@/types/assets'
+import { NetworkType } from '@/types/network'
+
+import client from '../client'
+
+export const assetsQueries = createQueryKeyStore({
+  type: {
+    details: (assetId: string) => ({
+      queryKey: [assetId],
+      queryFn: (): Promise<AssetBase> =>
+        client.node.guessStdTokenType(assetId).then((r) => ({ id: assetId, type: r ?? 'unknown' }))
+    })
+  },
+  metadata: {
+    allVerifiedTokens: (network: NetworkType) => ({
+      queryKey: [network],
+      queryFn: (): Promise<VerifiedFungibleTokenMetadata[]> =>
+        fetch(`https://raw.githubusercontent.com/alephium/token-list/master/tokens/${network}.json`).then((r) =>
+          r.json().then((j: TokenList) => j.tokens.map((v) => ({ ...v, type: 'fungible', verified: true })))
+        )
+    }),
```

Comment:
Unfortunately, the query factory library doesn't allow to configure the queries at this step. 
I agree that it would make sense to be able to provide a default behaviour here.

I thought about getting rid of this "key factory" thing and actually declare the full queries manually (grouped in file in the same way).

Yes we would need to "name" our tags ourselves, but we would have more flexibility to configure our api calls.

Reply:
I'd say that's a very small price to pay. 👍

Reply:
So I'll do that: get rid of this lib and declare the pre-configured queries in dedicated files.

Reply:
Done